Sources/TuistAutomation/Utilities/AppBundle.swift

```diff
@@ -29,27 +29,53 @@ public struct AppBundle: Equatable {
     }
 
     public struct InfoPlist: Codable, Equatable {
-        public struct PrimaryBundleIcon: Codable, Equatable {
+        public enum PrimaryBundleIcon: Codable, Equatable {
```

Comment:
I would keep this as a struct and leverage the fact that we already have custom encoding/decoding in place to handle the string/dictionary gracefully.

If the value of `PrimaryBundleIcon` is a string, we can initialize the struct with `name` set as that value and `iconFiles` as an empty array. How does that sound?

Reply:
> If the value of `PrimaryBundleIcon` is a string, we can initialize the struct with `name` set as that value and `iconFiles` as an empty array. How does that sound?

Yeah, I thought about that, but whether it's required to be `Encodable` I think it's might have an issue to detect whether `name` should be encoded within dictionary or not.